R/epidist_utils.R

```diff
@@ -0,0 +1,703 @@
+#' the citation
+#'
+#' @return A character string of the formatted short citation
+#' @export
+#'
+#' @examples
+#' create_epidist_citation(
+#'   author = "Smith_etal",
+#'   year = 2002,
+#'   DOI = "10.19832/j.1366-9516.2012.09147.x"
+#' )
+create_epidist_citation <- function(author = NA_character_,
+                                    year = NA_integer_,
+                                    PMID = NA_character_,
+                                    DOI = NA_character_,
+                                    use_PMID = FALSE) {
+  # check input
+  checkmate::assert_character(author)
+  checkmate::assert_number(year, na.ok = TRUE)
+  checkmate::assert_number(PMID, na.ok = TRUE)
+  checkmate::assert_character(DOI)
+  checkmate::assert_logical(use_PMID, len = 1)
+
+  if (is.na(author) || is.na(year) || is.na(DOI)) {
+    message(
+      "Citation cannot be created as either author, year or DOI is missing"
+    )
+    return("No citation available")
+  }
+
+  if (isTRUE(use_PMID) && is.na(PMID)) {
+    stop("use_PMID set to TRUE but PMID not provided")
+  }
+
+  # change author formatting if multiple authors or et al
+  author <- gsub(
+    pattern = "_",
+    replacement = " ",
+    x = author,
+    fixed = TRUE
+  )
+  author <- gsub(
+    pattern = "etal",
+    replacement = "et al.",
+    x = author,
+    fixed = TRUE
+  )
+
+  citation <- paste0(author, " (", year, ") ", "<", DOI, ">")
+
+  if (use_PMID) {
+    citation <- paste(citation, "PMID:", PMID)
+  }
+
+  citation
+}
+
+#' Specify whether certain methodological aspects have been used when fitting
+#' the distribution
+#'
+#' @description A helper function when creating an epidist object to create a method
+#' assessment list with sensible defaults, type checking and arguments to help
+#' remember which method assessments can be accepted in the list
+#'
+#' @details Currently, the method assessment focuses on common methodological
+#' aspects of delay distributions (e.g. incubation period, serial interval,
+#' etc.), and does not currently take into account methodological aspects which
+#' may be important when fitting offspring distributions to data on disease
+#' (super)spreading.
+#'
+#' @param censored A boolean logical whether the study used single or double
+#' interval censoring in the methods to infer the delay distribution
+#' @param right_truncated A boolean logical whether the study used right-
+#' truncation in the methods to infer the delay distribution
+#' @param phase_bias_adjusted A boolean logical whether the study adjusted for
+#' phase bias in the methods to infer the delay distribution
+#'
+#' @return A named list with three elements
+#' @export
+#'
+#' @examples
+#' create_epidist_method_assessment(
+#'   censored = FALSE,
+#'   right_truncated = FALSE,
+#'   phase_bias_adjusted = FALSE
+#' )
+create_epidist_method_assessment <- function(censored = NA,
+                                             right_truncated = NA,
```

Comment:
Could also be left truncated?

Reply:
From the papers used to collated parameters for the database I have not seen any report left truncation. However, I guess it is feasible that the delay distributions could be left-truncated so I will think about how best to implement this change. `left_truncated` can be easily added to `create_epidist_method_assessment()` but there are knock-on effects as there is currently only one column for truncation (assumed right truncation) in the database, so adding left truncation will also cause a change to the database.

Reply:
Serial intervals are often left truncated at 0 (implicitly excluding cases where A infects B but B develops symptoms before A). Here's an example where they were left truncated at 2 (cases with symptom onset 0 or 1 days apart were classified as co-primary): https://www.eurosurveillance.org/content/10.2807/1560-7917.ES.2022.27.15.2001551